yt/visualization/plot_container.py

```diff
@@ -894,8 +895,24 @@ def set_zlim(self, field, zmin, zmax, dynamic_range=None):
             zmin = zmax / dynamic_range.
 
         """
-        myzmin = zmin
-        myzmax = zmax
+
+        def _sanitize_units(z):
+            # convert dimensionful inputs to float
+            if isinstance(z, tuple):
+                z = YTQuantity(*z)
+            if isinstance(z, YTQuantity):
+                try:
+                    plot_units = self.frb[field].units
+                    z = z.to(plot_units).value
+                except AttributeError:
+                    # only certain subclasses have a frb attribute they can rely on for inspecting units
+                    from yt.funcs import mylog
```

Comment:
I think it's safe to put this in the main imports. I usually only do in-function imports if it's an optionally-installed module or something like that.

Reply:
Thanks for this rule of thumb, I'll try to keep this in mind :)